### PR TITLE
Upgrade libevent to 2.1.12-stable

### LINKED ISSUE #61: Update to libevent-2.1.12-stable
Latest version of libevent includes many fixes (see [Changelog](https://raw.githubusercontent.com/libevent/libevent/release-2.1.12-stable/ChangeLog)), but I am unable to build it because of missing OpenSSL includes for some reason:

```...
checking for openssl/ssl.h... no
configure: error: openssl is a must but can not be found. You should add the directory containing `openssl.pc' to the `PKG_CONFIG_PATH' environment variable, or set `CFLAGS' and `LDFLAGS' directly for openssl, or use `--disable-openssl' to disable support for openssl encryption
The command '/bin/sh -c curl -fsSL "https://github.com/libevent/libevent/releases/download/release-$libevent_version/libevent-$libevent_version.tar.gz" -o libevent-$libevent_version.tar.gz && curl -fsSL "https://github.com/libevent/libevent/releases/download/release-$libevent_version/libevent-$libevent_version.tar.gz.asc" -o libevent-$libevent_version.tar.gz.asc && gpg --import /libevent.gpg && gpg libevent-$libevent_version.tar.gz.asc && echo "$libevent_hash libevent-$libevent_version.tar.gz" | shasum -a 256 -c - && tar -zxvf libevent-$libevent_version.tar.gz && cd libevent-$libevent_version && ./configure LDFLAGS="-L$PWD/../openssl-$openssl_version/root" CPPFLAGS="-I$PWD/../oppenssl-$openssl_version/include" --prefix=$PWD/install --disable-shared --enable-static --with-pic --host=i686-w64-mingw32 && make ${jobs:+-j$jobs} && make ${jobs:+-j$jobs} install' returned a non-zero code: 1
```

Help appreciated :)

### PR DESCRIPTION
**Pull request by waybackarchiver**
Resolve #61
Related #34

### CODE REVIEW

**Comment by Mohsen7s** [comment]
Can somebody please approve this for the god sake!

File: build_darwin_x86_64.sh [hunk 1]
```diff
@@ -47,10 +47,9 @@ gpg --import gpg-keys/libevent.gpg && \
 gpg "libevent-$LIBEVENT_VERSION.tar.gz.asc" && \
 echo "$LIBEVENT_HASH  libevent-$LIBEVENT_VERSION.tar.gz" | shasum -a 256 -c - && \
 tar -zxvf "libevent-$LIBEVENT_VERSION.tar.gz" -C x86_64 && \
-cp patch/libevent/test/regress_bufferevent.c "x86_64/libevent-$LIBEVENT_VERSION/test/regress_bufferevent.c" && \
```
Comment (wknapik) [inline build_darwin_x86_64.sh:50]: If we're dropping the patch on macos x64, we should probably also drop it on macos arm64 and remove patch/libevent/test/regress_bufferevent.c from the repo, no?

**Comment by diracdeltas** [comment]
should be ok to update https://gitlab.torproject.org/tpo/applications/tor-browser-build/-/issues/40064

File: build_darwin_x86_64.sh (hunk 1 above)
Comment (diracdeltas) [reply build_darwin_x86_64.sh:50]: the macos builds are broken for unrelated reasons; once fixed in #78 i'll pull this into that PR

File: build_darwin_x86_64.sh [hunk 2]
```diff
@@ -47,10 +47,9 @@ gpg --import gpg-keys/libevent.gpg && \
...
 cd "x86_64/libevent-$LIBEVENT_VERSION" && \
 ./configure \
-            LDFLAGS="-L$PWD/../openssl-$OPENSSL_VERSION/root" \
+            LDFLAGS="-L$PWD/../openssl-$OPENSSL_VERSION/root/lib" \
```
Comment (diracdeltas) [inline build_darwin_x86_64.sh:52]: @waybackarchiver what's the reason for this change?

Comment (diracdeltas) [reply build_darwin_x86_64.sh:52]: i'm guessing it's https://github.com/libevent/libevent/issues/1136#issuecomment-792048863; probably needs to be updated in the win/linux dockerfiles as well

**Comment by diracdeltas** [comment]
i have pulled these changes cross-platform into #78 and am running a build to see if it works: https://ci.brave.com/blue/organizations/jenkins/brave-tor-client-build/detail/brave-tor-client-build/136/pipeline

**Comment by diracdeltas** [comment]
closing in favor of https://github.com/brave/tor_build_scripts/pull/78; more changes were needed, but thanks very much for starting this!

File: build_darwin_x86_64.sh (hunk 2 above)
Comment (waybackarchiver) [reply build_darwin_x86_64.sh:52]: Hmm yeah!
